modeling/gbert_classifier.py

```diff
@@ -0,0 +1,297 @@
+        self.bert = BertModel.from_pretrained("deepset/gbert-base")
+        self.dropout = nn.Dropout(p=0.3)
+        self.out = nn.Linear(self.bert.config.hidden_size, 1)
+
+    def forward(self, input_ids, attention_mask):
+        sequence_output, pooled_output = self.bert(
+          input_ids=input_ids,
+          attention_mask=attention_mask
+        ).values()
+        output = self.dropout(sequence_output)
+        output = self.out(output)
+        return torch.sigmoid(output)
+
+
+# %%
+def train_epoch(
+  model,
+  data_loader,
+  loss_fn,
+  optimizer,
+  device,
+  scheduler,
+  n_examples
+):
+    model = model.train()
+    losses = []
+    tp = 0
+    tn = 0
+    fp = 0
+    fn = 0
+    for d in data_loader:
+        input_ids = d["input_ids"].to(device)
+        attention_mask = d["attention_mask"].to(device)
+        targets = d["targets"].to(device)
+        outputs = model(
+          input_ids=input_ids,
+          attention_mask=attention_mask
+        )
+        preds = torch.round(outputs)
+        targets = targets.unsqueeze(1)
+        targets = targets.float()
+        loss = loss_fn(outputs, targets)
+        tp += (targets * preds).sum(dim=0).to(torch.float32)
+        tn += ((1 - targets) * (1 - preds)).sum(dim=0).to(torch.float32)
+        fp += ((1 - targets) * preds).sum(dim=0).to(torch.float32)
+        fn += (targets * (1 - preds)).sum(dim=0).to(torch.float32)
+        losses.append(loss.item())
+        loss.backward()
+        nn.utils.clip_grad_norm_(model.parameters(), max_norm=1.0)
+        optimizer.step()
+        scheduler.step()
+        optimizer.zero_grad()
+        div = tp + 0.5 * ( fp + fn)
+    return tp / div if div != 0 else 0, np.mean(losses)
+
+
+# %%
+def eval_model(model, data_loader, loss_fn, device, n_examples):
+    model = model.eval()
+    losses = []
+    tp = 0
+    tn = 0
+    fp = 0
+    fn = 0
+    with torch.no_grad():
+        for d in data_loader:
+            input_ids = d["input_ids"].to(device)
+            attention_mask = d["attention_mask"].to(device)
+            targets = d["targets"].to(device)
+            targets = targets.unsqueeze(1)
+            targets = targets.float()
+            outputs = model(
+                input_ids=input_ids,
+                attention_mask=attention_mask
+                )
+            preds = torch.round(outputs)
+            loss = loss_fn(outputs, targets)
+            tp += (targets * preds).sum(dim=0).to(torch.float32)
+            tn += ((1 - targets) * (1 - preds)).sum(dim=0).to(torch.float32)
+            fp += ((1 - targets) * preds).sum(dim=0).to(torch.float32)
+            fn += (targets * (1 - preds)).sum(dim=0).to(torch.float32)
+            losses.append(loss.item())
+            div = tp + 0.5 * ( fp + fn)
+    return tp / div if div != 0 else 0, np.mean(losses)
+
+
+# %% [markdown]
+# ## Data loading
+
+
+# %%
```

Comment:
I would move everything below in
```python
if __name__ == "__main__":
    ...
```
then it is only executed when run as a script/module:
```bash
python -m modeling.gberg_classifier
```